src/util.rs

```diff
@@ -12,3 +12,44 @@ pub fn is_audio_file(url: &CowStr) -> bool {
     }
     false
 }
+
+// return the extension of an url as a string
+pub fn get_ext(url: &CowStr) -> String {
```

Comment:
I think ideally this would avoid allocating a String, since we only use the extension to generate a mimetype, though ok to check in as is and iterate later :)

Reply:
cool.  This is my first Rust project, and I'd like to learn the most Rustic way to do things.  :)